### PR TITLE
CLIP-1500: Add security context for synchrony

### PR DESCRIPTION
**Pull request by bianchi2**
This PR adds security context for Synchrony (both in pod and container spec).

Fixes https://github.com/atlassian/data-center-helm-charts/issues/393

## Checklist
- [x] I have added unit tests
- [x] I have applied the change to all applicable products
- [x] I have added the change description to the `changelog.md` and `Chart.yaml` files
- [ ] (Atlassian only) I have run the E2E test (if applicable)
- [ ] (Atlassian only) Internal Bamboo CI is passing

### CODE REVIEW

File: src/main/charts/confluence/templates/statefulset-synchrony.yaml [hunk 1]
```diff
@@ -22,6 +22,20 @@ spec:
     spec:
       serviceAccountName: {{ include "confluence.serviceAccountName" . }}
       terminationGracePeriodSeconds: {{ .Values.synchrony.shutdown.terminationGracePeriodSeconds }}
+      {{- if .Values.confluence.securityContextEnabled }}
+      {{- with .Values.confluence.securityContext }}
```
Comment (nghazali) [inline src/main/charts/confluence/templates/statefulset-synchrony.yaml:26]: I am wonder if we should add a separated security context for Synchrony or we should re-use the confluence security context for Synchrony

Comment (bianchi2) [reply src/main/charts/confluence/templates/statefulset-synchrony.yaml:26]: I am not sure using different security contexts for Synchrony and Confluence is the use case we'll see much. That's why I decided to unify them not to bloat values.yaml.

File: src/main/charts/confluence/templates/statefulset-synchrony.yaml [hunk 2]
```diff
@@ -22,6 +22,20 @@ spec:
     spec:
       serviceAccountName: {{ include "confluence.serviceAccountName" . }}
       terminationGracePeriodSeconds: {{ .Values.synchrony.shutdown.terminationGracePeriodSeconds }}
+      {{- if .Values.synchrony.securityContextEnabled }}
+      {{- with .Values.synchrony.securityContext }}
+      securityContext:
+        {{/* this condition will be removed in v2.0.0 */}}
```
Comment (nanux) [inline src/main/charts/confluence/templates/statefulset-synchrony.yaml:28]: I don't think we need the condition here. We never had this defined for Synchrony statefulset so providing backward compatibility is not necessary.

File: src/main/charts/confluence/templates/statefulset-synchrony.yaml (hunk 1 above)
Comment (nanux) [reply src/main/charts/confluence/templates/statefulset-synchrony.yaml]: Thanks, @bianchi2. Based on our previous experience it is better to provide complete customizability for the user as it is hard to predict how they will use the helm charts. I agree that adding this to the helm chart is a good idea. In general, the guiding principle is to provide a high degree of flexibility and don't worry about the size of `values.yaml` file.

**Review by nanux — CHANGES_REQUESTED** [review]

File: src/main/charts/confluence/templates/statefulset-synchrony.yaml (hunk 2 above)
Comment (bianchi2) [reply src/main/charts/confluence/templates/statefulset-synchrony.yaml]: Thanks @nanux I wasn't quite sure why this condition was there, so decided not to delete it. Fixed now.

**Review by nanux — APPROVED** [review]

**Review by tarka — APPROVED** [review]

**Review by nghazali — APPROVED** [review]

**Pull request merged**